tests/v2/validation/provisioning/k3s/etcd_node_delete_test.go

```diff
@@ -0,0 +1,162 @@
+package k3s
+
+import (
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	nodestat "github.com/rancher/rancher/tests/framework/extensions/nodes"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	defaultNS                = "fleet-default"
+	etcdLabel                = "rke.cattle.io/etcd-role"
+	clusterLabel             = "cluster.x-k8s.io/cluster-name"
+	machineSteveResourceType = "cluster.x-k8s.io.machine"
+)
+
+type EtcdNodeDeleteTestSuite struct {
+	suite.Suite
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+}
+
+func (e *EtcdNodeDeleteTestSuite) TearDownSuite() {
+	e.session.Cleanup()
+}
+
+func (e *EtcdNodeDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	e.session = testSession
+
+	e.ns = defaultNS
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	e.k3skubernetesVersions = clustersConfig.K3SKubernetesVersions
+
+	e.cnis = clustersConfig.CNIs
+	e.providers = clustersConfig.Providers
+	e.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(e.T(), err)
+
+	e.client = client
+}
+
+func (e *EtcdNodeDeleteTestSuite) isEtcdNodeRemoved(query url.Values, clusterName string, etcdNodeToDelete v1.SteveAPIObject) (int, error) {
+	noOfEtcdNodesAfterDeletion := 0
+
+	err := kwait.Poll(5*time.Second, 15*time.Minute, func() (done bool, err error) {
+		machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
+		if err != nil {
+			return false, err
+		}
+
+		noOfEtcdNodesAfterDeletion = 0
+		for _, machine := range machines.Data {
+			if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
+				if machine.Name == etcdNodeToDelete.Name {
+					return false, nil
+				}
+				noOfEtcdNodesAfterDeletion++
```

Comment:
see comment below about `no` vs. `num` or `count`

Reply:
Change it to num.